neonvm/runner/containermgr.go

```diff
@@ -0,0 +1,364 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"go.uber.org/zap"
+
+	vmv1 "github.com/neondatabase/autoscaling/neonvm/apis/neonvm/v1"
+	"github.com/neondatabase/autoscaling/pkg/api"
+)
+
+const (
+	runnerContainerName = "neonvm-runner"
+
+	retryInitEvery = 5 * time.Second
+
+	// cpuLimitOvercommitFactor sets the amount above the VM's spec.guest.cpus.use that we set the
+	// QEMU cgroup's CPU limit to. e.g. if cpuLimitOvercommitFactor = 3 and the VM is using 0.5
+	// CPUs, we set the cgroup to limit QEMU+VM to 1.5 CPUs.
+	//
+	// This exists because setting the cgroup exactly equal to the VM's CPU value is overly
+	// pessimistic, results in a lot of unused capacity on the host, and particularly impacts
+	// operations that parallelize between the VM and QEMU, like heavy disk access.
+	//
+	// See also: https://neondb.slack.com/archives/C03TN5G758R/p1693462680623239
+	cpuLimitOvercommitFactor = 4
+
+	// number of CPU shares per vCPU to use when updating a container
+	sharesPerCPU = 1024
+	cpuPeriod    = 100000
+)
+
+// mainContainerMgr is the 'main' function if neonvm-runner is called with the container-mgr bin path
+//
+// This typically runs as a secondary container alongside neonvm-runner, with privileged access to
+// conatinerd via crictl.
+func mainContainerMgr() {
```

Comment:
Mostly to save on needing to distribute another image, but it's also nice to have the thing running alongside the runner be located alongside it in the code

Reply:
You can still ship both binaries on the same image and then override their entrypoint on the pod definition. I understand we want to keep the code next to each other but it feels we have coupled two separate things into the a single binary. An alternative is the following:
```
.
├── cmd/
│   ├── runner/
│   │   └── main.go
│   └── containermgr/
│       └── main.go
├── runner.go
└── containermgr.go
```

wdyt?

Reply:
Hm, yeah fair enough - I'll go with that.
Originally I was thinking it might be useful to share code between the two, but thinking about it further, I don't think we'll end up needing it.

Reply:
Done - ended up putting container-mgr stuff in a directory inside `runner`.

Reply:
Looks good. It's a bit unusual since the common convention is to put `main.go` files under the `cmd` directory but I don't have a strong incline towards it.